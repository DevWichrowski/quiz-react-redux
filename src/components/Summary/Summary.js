import React from 'react';
import "./Summary.scss";
import Header from "../shared/Header/Header";
import {connect} from "react-redux";
import {resetQuiz} from "../../store/actions/quiz.actions";
import {maxPoints} from "../../core/helpers/questionsCount";
import Button from "../shared/Button/Button";
import {Link, useHistory} from "react-router-dom";

const Summary = ({score, resetQuiz}) => {
    const history = useHistory();

    const retakeQuiz = () => {
        history.push("/quiz");
        resetQuiz();
    };

    return (
        <div className="summary">
            <Header headerText={"Summary"}/>
            <div className="summary__content">
                <div>
                    <h3>Thank you for participating in the quiz</h3>
                    <h2>Your score: {score}/{maxPoints}</h2>
                </div>
                <div className="buttons-container">
                    <Link to={"/"}>
                        <Button>Home</Button>
                    </Link>
                    <Button onClick={retakeQuiz} className="reset">Re-take quiz</Button>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = state => ({
    score: state.quiz.score,
});

const mapDispatchToProps = dispatch => ({
    resetQuiz: () => dispatch(resetQuiz())
});

export default connect(mapStateToProps, mapDispatchToProps)(Summary);