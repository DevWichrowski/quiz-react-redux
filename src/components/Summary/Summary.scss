@import "../../styles/colors";

.summary {
  height: 100%;

  &__content {
    width: 100%;
    height: 90%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    text-align: center;
    align-items: center;
    padding: 0 30px;

    h3 {
      font-size: 3vh;
    }

    h2 {
      font-size: 3.5vh;
      color: $green;
    }

    .buttons-container {
      display: flex;
      justify-content: center;
      width: 100%;
      margin-top: 40px;
      align-items: center;

      &__reset {
        margin-left: 10px;
        background-color: $gray;
        color: $blue;
      }

    }
  }
}