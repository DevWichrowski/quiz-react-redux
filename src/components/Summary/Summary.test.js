import {testConnectedComponentRender} from "../../core/helpers/tests/test-connected-component-render";
import Summary from "./Summary";


describe('Summary', () => {
    testConnectedComponentRender(Summary)
});
