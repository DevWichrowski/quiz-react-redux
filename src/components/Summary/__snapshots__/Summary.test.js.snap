// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary should render 1`] = `
<ContextProvider
  value={null}
>
  <Summary
    resetQuiz={[Function]}
    score={0}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  />
</ContextProvider>
`;
